src/service/search/cluster/mod.rs

```diff
@@ -621,19 +624,11 @@ async fn merge_grpc_result(
         // handle aggs
         for agg in resp.aggs {
             // insert count
-            if agg.name == "_count" && is_ingester(&node.role) {
-                let value = batches.entry("agg_ingester_count".to_string()).or_default();
-                if !agg.hits.is_empty() {
-                    let buf = Cursor::new(agg.hits.clone());
-                    let reader = ipc::reader::FileReader::try_new(buf, None).unwrap();
-                    let batch = reader
-                        .into_iter()
-                        .map(std::result::Result::unwrap)
-                        .collect::<Vec<_>>();
-                    value.extend(batch);
-                }
-            }
-            let value = batches.entry(format!("agg_{}", agg.name)).or_default();
+            let value = if agg.name == "_count" && node.role.contains(&Role::Ingester) {
```

Comment:
With `is_ingester(&node.role)`, in local mode, `agg_ingester_count` will always be added to `batches` regardless if the data has been indexed or not, causing the result to be inflated.

Reply:
`batches` different caused by this change:
Before:
`{"agg_ingester_count": val}` only present if the node is an ingester (`Role::Ingester` or `Role::All`)
`{"_count": val}` would always be in the batches

Now:
`{"agg_ingester_count": val}` only present if the node is an ingester (`Role::Ingester`)
`{"_count": val}` only present if "agg_ingester_count" entry not added to the `batches`. This is fine because below snippet in `src/service/search/cluster/http.rs` that `total` is assigned a value.
```rust
let total = match result.aggs.get("_count") {
    Some(v) => v.first().unwrap().get("num").unwrap().as_u64().unwrap() as usize,
    None => result.hits.len(),
};
```

Reply:
assigned a value but it is not a right value. `_count` and `hits.len()` is different things. the `hists.len()` just used to when the request don't need calculate real total just return the `hists.len` as a total.

Reply:
Okay, maybe this is not a problem, because we don't need `total` when we have `ingester_total`